pkg/cloudprovider/backup_service.go

```diff
@@ -145,12 +150,12 @@ func (br *backupService) UploadBackup(bucket, backupName string, metadata, backu
 	return nil
 }
 
-func (br *backupService) DownloadBackup(bucket, backupName string) (io.ReadCloser, error) {
-	return br.objectStorage.GetObject(bucket, getBackupContentsKey(backupName))
+func (br *backupService) DownloadBackup(bucket, bucketPath, backupName string) (io.ReadCloser, error) {
+	return br.objectStorage.GetObject(bucket, getBackupContentsKey(bucketPath, backupName))
 }
 
-func (br *backupService) GetAllBackups(bucket string) ([]*api.Backup, error) {
-	prefixes, err := br.objectStorage.ListCommonPrefixes(bucket, "/")
+func (br *backupService) GetAllBackups(bucket string, bucketPath string) ([]*api.Backup, error) {
+	prefixes, err := br.objectStorage.ListCommonPrefixes(bucket, "/", bucketPath)
```

Comment:
`bucketPath` needs to have a trailing `/`here if it's non-empty, otherwise ListCommonPrefixes ends up returning the `bucketPath` value rather than the backups contained within it.